pkg/storage/streams/store.go

```diff
@@ -167,7 +187,38 @@ func (s *streamStore) Get(ctx context.Context, path paths.Path) (
 			return nil, Meta{}, err
 		}
 
-		rangers = append(rangers, rangeCloser)
+		decrypter, err := eestream.NewAESGCMDecrypter(&encKey, &firstNonce, s.encryptedBlockSize)
+		if err != nil {
+			for _, ranger := range rangers {
+				_ = ranger.Close()
+			}
+			_ = lastRangerCloser.Close()
+			return nil, Meta{}, err
+		}
+
+		rd, err := eestream.Transform(rangeCloser, decrypter)
+		if err != nil {
+			for _, ranger := range rangers {
+				_ = ranger.Close()
+			}
+			_ = lastRangerCloser.Close()
+			return nil, Meta{}, err
+		}
+
+		paddedSize := decrypter.OutBlockSize()
+		size := int(rd.Size()) // int64 -> int; is this a problem?
+		rc, err := eestream.Unpad(rd, int(paddedSize-size))
+		if err != nil {
+			for _, ranger := range rangers {
```

Comment:
would it be better to `defer _ = ranger.Close()` every time we create a new ranger? That way we wouldn't need to manually close each ranger when we return.

Reply:
I don't know enough about Go yet to know whether that's a viable option

Reply:
We can't do this because this will close the ranger just before returning it to the caller. So the caller won't be able to read anything from it.

But we have to close the ranger if we are returning error, because the caller won't get it and cannot close it.